### PR TITLE
feat(analyzer): rework organise_file using pathlib

### PR DESCRIPTION
**Pull request by jooola**
The new behavior for duplicate file is now:
```
    If you import multiple copies of the same file, the behavior is:
    - The first filename is preserved.
    - The next filenames receive the current date append to the name.
```

- use datetime instead of uuids
- massively using pathlib to manipulate paths
- append datetime is now in iso format

Related to #1592

### CODE REVIEW

**Comment by jooola** [comment]
Please do not squash

File: analyzer/libretime_analyzer/pipeline/organise_file.py [hunk 1]
```diff
@@ -1,119 +1,56 @@
...
-    # the wrong information for the file we just overwrote (eg. the song length would be wrong!)
-    # If the final file path is the same as the file we've been told to import (which
-    # you often do when you're debugging), then don't move the file at all.
+    dest_path /= orig_filename.stem[0:MAX_FILE_LEN] + orig_filename.suffix
 
-    if os.path.exists(final_file_path):
-        if os.path.samefile(audio_file_path, final_file_path):
-            metadata["full_path"] = final_file_path
+    # Handle when a file already exists
+    while dest_path.is_file():
```
Comment (jooola) [inline analyzer/libretime_analyzer/pipeline/organise_file.py:39]: I don't think we will have to chance to generate a duplicate uuid4, but we never know !

**Comment by paddatrapper** [comment]
What are the chances of importing the same file twice in the same second? Would uuid not ensure uniqueness more reliably?

**Comment by jooola** [comment]
I initially replaced the date time suffix with an uuid, but then I thought we would loose information.

Using an UUID would make the code simpler, and theoretically safer. But the chances that 2 files with the same name (and with same metadata if any), are uploaded at the same second seem really low. The only possible scenario this could happen is if a bulk import or bulk scan a list of files with the same name, but in that case I expected the analyzer not to run more than 2 or 3 concurrent pipeline.

**Comment by paddatrapper** [comment]
The date is already in the database, so I think having a UUID here is better. The filenames themselves are not actually important in the grand scheme - everything displays things based on metadata in the DB anyway

**Comment by jooola** [comment]
Replaced datetime with uuids

**Pull request merged**